lms/services/lti_grading/_v11.py

```diff
@@ -49,22 +54,52 @@ def read_result(self, grading_id) -> GradingResult:
         return result
 
     def record_result(self, grading_id, score=None, pre_record_hook=None, comment=None):  # noqa: ARG002
-        request = {"resultRecord": {"sourcedGUID": {"sourcedId": grading_id}}}
-
-        if score is not None:
-            request["resultRecord"]["result"] = {
-                "resultScore": {"language": "en", "textString": score}
-            }
-
+        request = self._record_score_payload(score=score, user_grading_id=grading_id)
         if pre_record_hook:
             request = pre_record_hook(score=score, request_body=request)
 
-        self._send_request({"replaceResultRequest": request})
+        self._send_request(
+            self.application_instance,
+            self.line_item_url,
+            {"replaceResultRequest": request},
+        )
 
-    def _send_request(self, request_body) -> dict:
+    def sync_grade(
+        self,
+        application_instance: ApplicationInstance,
+        assignment,
+        _grade_timestamp: str,
+        lms_user: LMSUser,
+        score: float,
+    ):
+        assignment_membership: LMSUserAssignmentMembership = self.db.scalars(
+            select(LMSUserAssignmentMembership).where(
+                LMSUserAssignmentMembership.lms_user_id == lms_user.id,
+                LMSUserAssignmentMembership.assignment_id == assignment.id,
+            )
+        ).one()
+        assert (
+            assignment_membership.lti_v11_lis_result_sourcedid
+        ), "Trying to grade a student without lti_v11_lis_result_sourcedid"
+
+        request = self._record_score_payload(
+            score=score,
+            user_grading_id=assignment_membership.lti_v11_lis_result_sourcedid,
+        )
+        self._send_request(
+            application_instance,
```

Comment:
Here we call send_reuest with the passed parameters instead of the ones in the instance.